test/parallel_api/numeric/numeric.ops/histogram.pass.cpp

```diff
@@ -0,0 +1,131 @@
+        histogram_sequential(in_first, in_last, boundary_first, boundary_last, expected_bin_first);
+        auto orr = ::oneapi::dpl::histogram(exec, in_first, in_last, boundary_first, boundary_last, bin_first);
+        EXPECT_TRUE(bin_last == orr, "histogram returned wrong iterator");
+        EXPECT_EQ_N(expected_bin_first, bin_first, bin_size, "wrong result from histogram");
+        ::std::fill_n(bin_first, bin_size, trash);
+    }
+};
+
+template <typename Size, typename T>
+void
+test_range_and_even_histogram(Size n, T min_boundary, T max_boundary, T overflow, Size jitter, Size num_bins,
+                              Size trash)
+{
+    //possibly spill over by overflow/2 on each side of range
+    Sequence<T> in(n, [&](size_t k) {
+        return (std::rand() % Size(max_boundary - min_boundary + overflow)) + min_boundary - overflow / 2;
+    });
+
+    Sequence<Size> expected(num_bins, [](size_t k) { return 0; });
+    Sequence<Size> out(num_bins, [&](size_t k) { return trash; });
+
+    invoke_on_all_hetero_policies<0>()(test_histogram_even_bins(), in.begin(), in.end(), expected.begin(),
+                                       expected.end(), out.begin(), out.end(), Size(in.size()), min_boundary,
+                                       max_boundary, trash);
+#    if !ONEDPL_FPGA_DEVICE
+    invoke_on_all_hetero_policies<1>()(test_histogram_even_bins(), in.cbegin(), in.cend(), expected.begin(),
+                                       expected.end(), out.begin(), out.end(), Size(in.size()), min_boundary,
+                                       max_boundary, trash);
+#    endif // !ONEDPL_FPGA_DEVICE
+
+    T offset = (max_boundary - min_boundary) / T(num_bins);
+    Sequence<T> boundaries(num_bins + 1, [&](size_t k) { return k * offset + (std::rand() % jitter) + min_boundary; });
+
+    invoke_on_all_hetero_policies<2>()(test_histogram_range_bins(), in.begin(), in.end(), boundaries.begin(),
+                                       boundaries.end(), expected.begin(), expected.end(), out.begin(), out.end(),
+                                       trash);
+#    if !ONEDPL_FPGA_DEVICE
+    invoke_on_all_hetero_policies<3>()(test_histogram_range_bins(), in.cbegin(), in.cend(), boundaries.cbegin(),
+                                       boundaries.cend(), expected.begin(), expected.end(), out.begin(), out.end(),
+                                       trash);
+#    endif // !ONEDPL_FPGA_DEVICE
+}
+
+template <typename T, typename Size>
+void
+test_histogram(T min_boundary, T max_boundary, T overflow, Size jitter, Size trash)
+{
+    for (Size bin_size = 4; bin_size <= 20000; bin_size = Size(3.1415 * bin_size))
+    {
+        for (Size n = 0; n <= 100000; n = n <= 16 ? n + 1 : Size(3.1415 * n))
+        {
+            test_range_and_even_histogram(n, min_boundary, max_boundary, overflow, jitter, bin_size, trash);
+        }
+
+        // testing of large number of items may take too much time in debug mode
+        Size n =
```

Comment:
constexpr?

Reply:
Still see `Size n =` in diffs...

Reply:
hmm, make sure you are looking at all commits rather than just a subset... it is changed in this commit: https://github.com/oneapi-src/oneDPL/pull/1273/commits/5cc254c88860f476512e74877cb7d7b5cbb4605f